date-time/src/commonMain/kotlin/com/splendo/kaluga/datetime/timer/InstantTimer.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.datetime.timer
+
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlin.time.Duration
+
+/** This timer finishes instantly. */
+internal class InstantTimer(override val duration: Duration) : Timer {
```

Comment:
What is application of this object in real world app?

Reply:
my intention was to use it during development and in tests

Reply:
Maybe move to commonTest only? And clean TimerProvider also if it's not used

Reply:
Timer with 0 duration and 0 interval should behave as instantly finished one

Reply:
not really, it won't emit the target duration from `elapsed()`

Reply:
on a second thought, it's easy to implement if needed. removing...